cmd/proxy/main.go

```diff
@@ -20,14 +23,11 @@ func main() {
 		log.Fatal(err)
 	}
 
-	w := app.Worker
-	mf := module.NewFilter()
-	if err := w.Register(actions.FetcherWorkerName, actions.GetProcessCacheMissJob(s, w, mf)); err != nil {
-		log.Fatal(err)
-	}
-
-	if err := w.Register(actions.ReporterWorkerName, actions.GetCacheMissReporterJob(w, mf)); err != nil {
-		log.Fatal(err)
+	if w, ok := app.Worker.(*gwa.Adapter); ok {
+		registerWithOptions(w, s)
+	} else {
+		w := app.Worker
+		registerNoOptions(w, s)
```

Comment:
Couldn't we create the worker here and pass it to the app? We wouldn't need the type assertion. Just like in Olympus/main.go

Reply:
yes, but we have auth.go, I did not want to create it there and pass it down

Reply:
I don't think we need `registerNoOptions` at all no? We're always using `*gwa.Adapter` and since `actions.App()` doesn't accept any arguments, it's always gonna initialize an app with a redis worker. I'm gonna add a general comment in the review a little more